### PR TITLE
sql/spanner: Add initial ARRAY type handling

### PR DESCRIPTION
**Pull request by tmc**
This adds initial ARRAY type support to the spanner driver.

### CODE REVIEW

File: sql/spanner/convert.go [hunk 1]
```diff
@@ -94,7 +96,7 @@ func ParseType(c string) (schema.Type, error) {
 			T: TypeDate,
 		}, nil
 	case TypeBool:
-		return &schema.TimeType{
+		return &schema.BoolType{
```
Comment (tmc) [inline sql/spanner/convert.go:99]: note: this is an unrelated fixup

File: sql/spanner/inspect.go [hunk 2]
```diff
@@ -214,6 +217,19 @@ func columnType(spannerType string) (schema.Type, error) {
 	col := &columnDesc{}
 	spannerType = strings.TrimSpace(strings.ToUpper(spannerType))
 
+	// ARRAY type handling.
+	if strings.HasPrefix(spannerType, "ARRAY<") {
```
Comment (a8m) [inline sql/spanner/inspect.go:221]: You can also use `arrayTypeRe.MatchString(spannerType )`.

**Review by a8m — APPROVED** [review]
💯

File: sql/spanner/inspect.go (hunk 2 above)
Comment (tmc) [reply sql/spanner/inspect.go:221]: what's preferable?

**Comment by a8m** [comment]
@tmc, fix the merge conflict, and I'll merge this 🙏

**Comment by tmc** [comment]
@a8m pushed a fix to the test file resolving the conflict.
